File: public/gen-icons.js
// 아이콘 생성 스크립트 (Node.js + canvas)
const { createCanvas } = require("canvas");
const fs = require("fs");
const path = require("path");

const SIZES = [72, 96, 128, 144, 152, 192, 384, 512];
const OUT = path.join(__dirname, "icons");
if (!fs.existsSync(OUT)) fs.mkdirSync(OUT, { recursive: true });

function drawIcon(size) {
  const c = createCanvas(size, size);
  const ctx = c.getContext("2d");
  const s = size;

  // Background
  const bg = ctx.createLinearGradient(0, 0, s, s);
  bg.addColorStop(0, "#0d1a28");
  bg.addColorStop(1, "#1a0a00");
  ctx.fillStyle = bg;
  ctx.beginPath();
  ctx.roundRect(0, 0, s, s, s * 0.18);
  ctx.fill();

  // Green glow circle (ChatGPT aura)
  const glow = ctx.createRadialGradient(s * 0.35, s * 0.5, 0, s * 0.35, s * 0.5, s * 0.35);
  glow.addColorStop(0, "rgba(16,163,127,0.35)");
  glow.addColorStop(1, "rgba(16,163,127,0)");
  ctx.fillStyle = glow;
  ctx.beginPath();
  ctx.arc(s * 0.35, s * 0.5, s * 0.35, 0, Math.PI * 2);
  ctx.fill();

  // Boss body (right side, angry red face)
  ctx.fillStyle = "#6677aa";
  ctx.beginPath();
  ctx.roundRect(s * 0.5, s * 0.42, s * 0.38, s * 0.3, s * 0.06);
  ctx.fill();
  // Boss head (red angry)
  ctx.fillStyle = "#e84040";
  ctx.beginPath();
  ctx.ellipse(s * 0.69, s * 0.34, s * 0.17, s * 0.15, 0, 0, Math.PI * 2);
  ctx.fill();
  ctx.strokeStyle = "#a02010";
  ctx.lineWidth = s * 0.015;
  ctx.stroke();
  // Boss angry eyes
  ctx.fillStyle = "#fff";
  ctx.beginPath(); ctx.ellipse(s * 0.63, s * 0.32, s * 0.04, s * 0.03, 0, 0, Math.PI * 2); ctx.fill();
  ctx.beginPath(); ctx.ellipse(s * 0.75, s * 0.32, s * 0.04, s * 0.03, 0, 0, Math.PI * 2); ctx.fill();
  ctx.fillStyle = "#111";
  ctx.beginPath(); ctx.ellipse(s * 0.63, s * 0.32, s * 0.02, s * 0.02, 0, 0, Math.PI * 2); ctx.fill();
  ctx.beginPath(); ctx.ellipse(s * 0.75, s * 0.32, s * 0.02, s * 0.02, 0, 0, Math.PI * 2); ctx.fill();
  // Boss V-brows
  ctx.strokeStyle = "#111"; ctx.lineWidth = s * 0.018; ctx.lineCap = "round";
  ctx.beginPath(); ctx.moveTo(s * 0.59, s * 0.26); ctx.lineTo(s * 0.65, s * 0.29); ctx.stroke();
  ctx.beginPath(); ctx.moveTo(s * 0.71, s * 0.29); ctx.lineTo(s * 0.77, s * 0.26); ctx.stroke();
  // Boss shout mouth
  ctx.fillStyle = "#111";
  ctx.beginPath(); ctx.ellipse(s * 0.69, s * 0.38, s * 0.06, s * 0.04, 0, 0, Math.PI * 2); ctx.fill();

  // ChatGPT robot (left side)
  // Robot body
  const robGrad = ctx.createLinearGradient(s * 0.05, s * 0.4, s * 0.42, s * 0.85);
  robGrad.addColorStop(0, "#1e3040");
  robGrad.addColorStop(1, "#0d1e2e");
  ctx.fillStyle = robGrad;
  ctx.beginPath();
  ctx.roundRect(s * 0.05, s * 0.48, s * 0.37, s * 0.38, s * 0.07);
  ctx.fill();
  ctx.strokeStyle = "#10a37f"; ctx.lineWidth = s * 0.02;
  ctx.stroke();
  // Robot head
  ctx.fillStyle = "#1a2a3a";
  ctx.beginPath();
  ctx.roundRect(s * 0.09, s * 0.24, s * 0.29, s * 0.26, s * 0.06);
  ctx.fill();
  ctx.strokeStyle = "#10a37f"; ctx.lineWidth = s * 0.018;
  ctx.stroke();
  // Robot antenna
  ctx.strokeStyle = "#10a37f"; ctx.lineWidth = s * 0.015;
  ctx.beginPath(); ctx.moveTo(s * 0.235, s * 0.24); ctx.lineTo(s * 0.235, s * 0.14); ctx.stroke();
  ctx.fillStyle = "#10a37f";
  ctx.beginPath(); ctx.arc(s * 0.235, s * 0.12, s * 0.025, 0, Math.PI * 2); ctx.fill();
  // Robot eyes (glowing green)
  const eyeGlow1 = ctx.createRadialGradient(s * 0.165, s * 0.35, 0, s * 0.165, s * 0.35, s * 0.05);
  eyeGlow1.addColorStop(0, "#80ffcc");
  eyeGlow1.addColorStop(1, "#10a37f");
  ctx.fillStyle = eyeGlow1;
  ctx.beginPath(); ctx.ellipse(s * 0.165, s * 0.35, s * 0.05, s * 0.04, 0, 0, Math.PI * 2); ctx.fill();
  const eyeGlow2 = ctx.createRadialGradient(s * 0.305, s * 0.35, 0, s * 0.305, s * 0.35, s * 0.05);
  eyeGlow2.addColorStop(0, "#80ffcc");
  eyeGlow2.addColorStop(1, "#10a37f");
  ctx.fillStyle = eyeGlow2;
  ctx.beginPath(); ctx.ellipse(s * 0.305, s * 0.35, s * 0.05, s * 0.04, 0, 0, Math.PI * 2); ctx.fill();
  // Robot mouth LED
  ctx.fillStyle = "#0a1a14";
  ctx.beginPath(); ctx.roundRect(s * 0.14, s * 0.43, s * 0.19, s * 0.04, s * 0.01); ctx.fill();
  ctx.fillStyle = "#10a37f";
  for (let i = 0; i < 4; i++) {
    ctx.beginPath(); ctx.arc(s * (0.155 + i * 0.05), s * 0.45, s * 0.012, 0, Math.PI * 2); ctx.fill();
  }
  // Robot chest screen
  ctx.fillStyle = "#0a1a14";
  ctx.beginPath(); ctx.roundRect(s * 0.1, s * 0.55, s * 0.27, s * 0.16, s * 0.04); ctx.fill();
  ctx.strokeStyle = "#10a37f"; ctx.lineWidth = s * 0.012; ctx.stroke();
  // Charge bar inside screen
  ctx.fillStyle = "#10a37f";
  ctx.beginPath(); ctx.roundRect(s * 0.115, s * 0.595, s * 0.17, s * 0.04, s * 0.01); ctx.fill();
  ctx.fillStyle = "#00ffcc";
  ctx.font = `bold ${Math.floor(s * 0.07)}px Arial`;
  ctx.textAlign = "center";
  ctx.fillText("100%", s * 0.235, s * 0.65);

  // Beam from robot to boss
  ctx.save();
  ctx.globalAlpha = 0.85;
  // outer glow
  ctx.strokeStyle = "rgba(16,163,127,0.25)"; ctx.lineWidth = s * 0.06;
  ctx.beginPath(); ctx.moveTo(s * 0.42, s * 0.52); ctx.lineTo(s * 0.52, s * 0.38); ctx.stroke();
  // beam core
  ctx.strokeStyle = "#10a37f"; ctx.lineWidth = s * 0.025;
  ctx.beginPath(); ctx.moveTo(s * 0.42, s * 0.52); ctx.lineTo(s * 0.52, s * 0.38); ctx.stroke();
  ctx.strokeStyle = "#ffffff"; ctx.lineWidth = s * 0.008;
  ctx.beginPath(); ctx.moveTo(s * 0.42, s * 0.52); ctx.lineTo(s * 0.52, s * 0.38); ctx.stroke();
  ctx.restore();

  // Hit sparks on boss
  const sparkCols = ["#ffcc00", "#ff7700", "#ffffff"];
  [[s*0.58, s*0.3],[s*0.72, s*0.22],[s*0.8, s*0.35]].forEach(([sx, sy], i) => {
    ctx.fillStyle = sparkCols[i % sparkCols.length];
    drawStar(ctx, sx, sy, s * 0.04, 5);
  });

  // Title text at bottom
  ctx.fillStyle = "#ffcc00";
  ctx.font = `bold ${Math.floor(s * 0.08)}px Arial`;
  ctx.textAlign = "center";
  ctx.fillText("상사혼내기", s * 0.5, s * 0.94);

  return c.toBuffer("image/png");
}

function drawStar(ctx, cx, cy, r, pts) {
  ctx.beginPath();
  for (let i = 0; i < pts * 2; i++) {
    const a = i * Math.PI / pts - Math.PI / 2;
    const rad = i % 2 === 0 ? r : r * 0.42;
    const px = cx + Math.cos(a) * rad, py = cy + Math.sin(a) * rad;
    i === 0 ? ctx.moveTo(px, py) : ctx.lineTo(px, py);
  }
  ctx.closePath();
  ctx.fill();
}

for (const size of SIZES) {
  const buf = drawIcon(size);
  const fp = path.join(OUT, `icon-${size}.png`);
  fs.writeFileSync(fp, buf);
  console.log(`✓ icon-${size}.png`);
}

// also make a simple screenshot placeholder
const sc = createCanvas(390, 844);
const sctx = sc.getContext("2d");
const sbg = sctx.createLinearGradient(0, 0, 390, 844);
sbg.addColorStop(0, "#0d1a28"); sbg.addColorStop(1, "#1a0a00");
sctx.fillStyle = sbg; sctx.fillRect(0, 0, 390, 844);
sctx.fillStyle = "#ffcc00";
sctx.font = "bold 36px Arial"; sctx.textAlign = "center";
sctx.fillText("직장상사 혼내기!", 195, 400);
sctx.fillStyle = "#10a37f";
sctx.font = "bold 22px Arial";
sctx.fillText("with ChatGPT", 195, 440);
fs.writeFileSync(path.join(OUT, "screenshot.png"), sc.toBuffer("image/png"));
console.log("✓ screenshot.png");
console.log("아이콘 생성 완료!");

File: public/gen-icons.py
#!/usr/bin/env python3
"""아이콘 생성 스크립트 (Pillow)"""
import math, os
from PIL import Image, ImageDraw, ImageFont

SIZES = [72, 96, 128, 144, 152, 192, 384, 512]
OUT = os.path.join(os.path.dirname(__file__), "icons")
os.makedirs(OUT, exist_ok=True)

def lerp_color(c1, c2, t):
    return tuple(int(c1[i] + (c2[i]-c1[i])*t) for i in range(3))

def draw_rounded_rect(draw, box, radius, fill, outline=None, width=1):
    x0,y0,x1,y1 = box
    draw.rounded_rectangle([x0,y0,x1,y1], radius=radius, fill=fill, outline=outline, width=width)

def draw_star(draw, cx, cy, r, pts, fill):
    verts = []
    for i in range(pts*2):
        a = i*math.pi/pts - math.pi/2
        rad = r if i%2==0 else r*0.42
        verts.append((cx+math.cos(a)*rad, cy+math.sin(a)*rad))
    draw.polygon(verts, fill=fill)

def make_icon(size):
    img = Image.new("RGBA", (size, size), (0,0,0,0))
    d = ImageDraw.Draw(img)
    s = size

    # ── Background gradient (simulate with two rects) ──
    for y in range(s):
        t = y/s
        c = lerp_color((13,26,40),(26,10,0),t)
        d.line([(0,y),(s,y)], fill=c+(255,))
    # rounded mask
    mask = Image.new("L",(s,s),0)
    md = ImageDraw.Draw(mask)
    md.rounded_rectangle([0,0,s,s], radius=int(s*.18), fill=255)
    img.putalpha(mask)
    d = ImageDraw.Draw(img)

    # ── Green glow (ChatGPT) ──
    gx,gy,gr = int(s*.35),int(s*.5),int(s*.32)
    for r in range(gr,0,-2):
        a = int(80*(1-r/gr))
        d.ellipse([gx-r,gy-r,gx+r,gy+r], fill=(16,163,127,a))

    # ── Boss (right, angry red) ──
    # Body
    draw_rounded_rect(d,[int(s*.5),int(s*.42),int(s*.88),int(s*.72)],int(s*.06),fill=(102,119,170,230),outline=(68,85,136),width=max(1,int(s*.015)))
    # Suit lapels
    d.polygon([(int(s*.5),int(s*.5)),(int(s*.62),int(s*.42)),(int(s*.65),int(s*.64))],fill=(119,136,187,200))
    d.polygon([(int(s*.88),int(s*.5)),(int(s*.76),int(s*.42)),(int(s*.73),int(s*.64))],fill=(119,136,187,200))
    # Head (red angry)
    d.ellipse([int(s*.52),int(s*.18),int(s*.88),int(s*.44)],fill=(232,64,64,240),outline=(160,32,16),width=max(1,int(s*.015)))
    # Hair
    d.ellipse([int(s*.55),int(s*.14),int(s*.86),int(s*.26)],fill=(42,34,16,230))
    # Eyes
    d.ellipse([int(s*.57),int(s*.27),int(s*.66),int(s*.35)],fill=(255,255,255))
    d.ellipse([int(s*.71),int(s*.27),int(s*.8),int(s*.35)],fill=(255,255,255))
    d.ellipse([int(s*.60),int(s*.29),int(s*.64),int(s*.33)],fill=(17,17,17))
    d.ellipse([int(s*.74),int(s*.29),int(s*.78),int(s*.33)],fill=(17,17,17))
    # Angry V-brows
    lw = max(2,int(s*.018))
    d.line([int(s*.57),int(s*.24),int(s*.64),int(s*.28)],fill=(17,17,17),width=lw)
    d.line([int(s*.73),int(s*.28),int(s*.80),int(s*.24)],fill=(17,17,17),width=lw)
    # Open mouth
    d.ellipse([int(s*.63),int(s*.36),int(s*.76),int(s*.44)],fill=(17,17,17))
    d.ellipse([int(s*.64),int(s*.37),int(s*.75),int(s*.43)],fill=(204,34,34))

    # ── ChatGPT Robot (left) ──
    # Robot body
    draw_rounded_rect(d,[int(s*.06),int(s*.48),int(s*.43),int(s*.86)],int(s*.07),fill=(30,48,64,240),outline=(16,163,127),width=max(1,int(s*.02)))
    # Robot head
    draw_rounded_rect(d,[int(s*.09),int(s*.22),int(s*.41),int(s*.48)],int(s*.06),fill=(26,42,58,240),outline=(16,163,127),width=max(1,int(s*.018)))
    # Antenna
    lw2 = max(2,int(s*.015))
    d.line([int(s*.245),int(s*.22),int(s*.245),int(s*.13)],fill=(16,163,127),width=lw2)
    d.ellipse([int(s*.22),int(s*.10),int(s*.27),int(s*.15)],fill=(16,163,127))
    d.ellipse([int(s*.225),int(s*.105),int(s*.265),int(s*.145)],fill=(100,255,200))
    # Eyes glow
    ey = int(s*.34)
    for ex in [int(s*.165),int(s*.31)]:
        d.ellipse([ex-int(s*.055),ey-int(s*.04),ex+int(s*.055),ey+int(s*.04)],fill=(10,26,20))
        d.ellipse([ex-int(s*.045),ey-int(s*.032),ex+int(s*.045),ey+int(s*.032)],fill=(16,163,127))
        d.ellipse([ex-int(s*.025),ey-int(s*.02),ex+int(s*.025),ey+int(s*.02)],fill=(100,255,200))
    # Mouth LED
    draw_rounded_rect(d,[int(s*.14),int(s*.42),int(s*.36),int(s*.47)],int(s*.01),fill=(10,26,20))
    for i in range(4):
        mx = int(s*(.16+i*.05))
        d.ellipse([mx-int(s*.012),int(s*.435),mx+int(s*.012),int(s*.46)],fill=(16,163,127))
    # Chest screen
    draw_rounded_rect(d,[int(s*.1),int(s*.53),int(s*.39),int(s*.7)],int(s*.04),fill=(10,26,20,220),outline=(16,163,127),width=max(1,int(s*.012)))
    # Charge bar
    draw_rounded_rect(d,[int(s*.115),int(s*.565),int(s*.375),int(s*.605)],int(s*.01),fill=(16,163,127))
    # "100%" text
    try:
        fsize = max(8,int(s*.07))
        font = ImageFont.truetype("/usr/share/fonts/truetype/dejavu/DejaVuSans-Bold.ttf", fsize)
    except:
        font = ImageFont.load_default()
    d.text((int(s*.245),int(s*.615)),"100%",fill=(0,255,180),font=font,anchor="mm" if hasattr(font,"getbbox") else None)

    # ── Beam ──
    bx0,by0 = int(s*.42),int(s*.52)
    bx1,by1 = int(s*.52),int(s*.38)
    for w,a in [(int(s*.07),40),(int(s*.03),140),(int(s*.01),220)]:
        col = (16,163,127,a) if w>int(s*.02) else (255,255,255,a)
        d.line([bx0,by0,bx1,by1],fill=col,width=w)

    # ── Sparks on boss ──
    spark_positions = [(int(s*.6),int(s*.3)),(int(s*.74),int(s*.21)),(int(s*.82),int(s*.35))]
    spark_colors = [(255,204,0,220),(255,119,0,200),(255,255,255,220)]
    for (sx,sy),sc in zip(spark_positions,spark_colors):
        draw_star(d,sx,sy,int(s*.045),5,sc)

    # ── Title text ──
    try:
        tfont = ImageFont.truetype("/usr/share/fonts/truetype/dejavu/DejaVuSans-Bold.ttf", max(8,int(s*.075)))
    except:
        tfont = ImageFont.load_default()
    d.text((s//2, int(s*.94)), "상사혼내기", fill=(255,204,0,240), font=tfont,
           anchor="mm" if hasattr(tfont,"getbbox") else None)

    return img

for size in SIZES:
    img = make_icon(size)
    fp = os.path.join(OUT, f"icon-{size}.png")
    img.save(fp, "PNG")
    print(f"✓ icon-{size}.png")

# screenshot placeholder
sc = Image.new("RGB",(390,844),(13,26,40))
sd = ImageDraw.Draw(sc)
for y in range(844):
    t=y/844
    c=lerp_color((13,26,40),(26,10,0),t)
    sd.line([(0,y),(390,y)],fill=c)
try:
    tf=ImageFont.truetype("/usr/share/fonts/truetype/dejavu/DejaVuSans-Bold.ttf",32)
    sf=ImageFont.truetype("/usr/share/fonts/truetype/dejavu/DejaVuSans-Bold.ttf",20)
except:
    tf=sf=ImageFont.load_default()
sd.text((195,390),"직장상사 혼내기!",fill=(255,204,0),font=tf,anchor="mm")
sd.text((195,440),"with ChatGPT",fill=(16,163,127),font=sf,anchor="mm")
sc.save(os.path.join(OUT,"screenshot.png"),"PNG")
print("✓ screenshot.png")
print("아이콘 생성 완료!")
